content/blogs/2023-05-30-track-resource-usage-from-bigquery-with-kestra-metric-dashboard

```diff
@@ -0,0 +1,77 @@
+---
+title: "Track Resource Usage from BigQuery with Kestra's Metric Dashboard"
+description: Get insight on the bytes processed by your queries on BigQuery to manage your costs. 
+date: 2023-05-30T12:00:00
+category: Solutions
+author:
+  name: Benoit Pimpaud
+  image: "bpimpaud"
+image: /blogs/2023-05-30-track-resource-usage-from-bigquery-with-kestra-metric-dashboard.jpg
+---
+
+Data Warehouse solutions can sometimes become costly, especially with an uptick in data processing demand. Cloud-based data warehouses like Google BigQuery offer significant advantages in terms of scalability and performance. Yet, with great scalability comes great responsibility, managing your ressources usage effectively. 
+
+## Track BigQuery usage ##
+
+BigQuery is very fast and easy to use. But it can be very expensive, especially when running scheduled queries : you’re not here to look at the bytes billed every time you’re query run.
+
+In the example below, we create a Kestra task to run a query on a Big Query table. To schedule our query, we just add a Kestra Trigger with a cron property allowing us to run the query every hour.
+
+```yaml
+id: query_events
+namespace: sales
+
+tasks:
+
+  - id: query
+    type: io.kestra.plugin.gcp.bigquery.Query
+    sql: |
+      SELECT 
+        SUM(price) as sum_price 
+      FROM sales.event_partitioned
+      WHERE date = '{{ schedule.date }}'
+    fetch: true
+
+triggers:
+  - id: schedule
+    type: io.kestra.core.models.triggers.types.Schedule
+    cron: "0 * * * *
+```
+
+### Flow Code Breakdown ###
+
+- First, we establish a task, identified as "query_events", in the "sales" namespace.
+
+- The task contains a single task (or step), identified as "query". The type of this task indicates it uses the BigQuery plugin to execute SQL queries.
+
+- The SQL query is structured to select the sum of a 'price' column from a BigQuery table named 'sales.event_partitioned', with the condition that the date equals the scheduled date.
+
+- The 'fetch' parameter is set to 'true', indicating that Kestra will retrieve and store the query results.
+
+- The task also includes a 'schedule' trigger. This is a type of Schedule trigger, which uses a cron format string to schedule when the task will be executed. The cron string "0 * * * *" indicates that the task will run every hour, on the hour.
+
+So, Kestra will execute this task every hour, running the specified SQL query on the BigQuery 'sales.event_partitioned' table, for the current date, and store the result. This Query allow to get the total bytes processed, the query duration or the number of referenced tables, and more.
```

Comment:
> This Query allow to get the total bytes processed, the query duration or the number of referenced tables, and more.

either : 

👉 this query allows to retrieve the total sale prices for a given day.
👉 The BigQuery Query task expose metrics such as the total bytes processed, the query duration or the number of referenced tables, and more.